Generate snippets file for Visual Studio Code

ekoleda-codaio opened the pull request:
Updates the documentation compiler to produce a `.code-snippets` file that is compatible with VSCode, and updates `coda init` to copy that file into the project's `.vscode` folder. Some design choices:

- Trigger tokens are prefixed with a forward slash to mirror the Pack Studio. This is not required in VSCode, but felt like it was better for consistency.
- The `pack.code-snippets` file is copied once during init, so updating the SDK version won't automatically bring in the latest snippets. I feel like this OK, but perhaps symlinks are an option?

![image](https://github.com/coda/packs-sdk/assets/88106038/bd920278-c9ec-4bce-ac88-441c81ac86c2)

jonathan-codaio reviewed (commented):
Awesome! I was going to ask if this was a possibility, and here it is...

Kinda seems like adding the VSCode snippets should be its own command, so you can do it whenever you want including after an SDK update, and do it independently of `coda init`. Fine if `coda init` also does it on your behalf (though there's an argument that if you don't use VSCode you'll have this weird artifact that you'll need to .gitignore, though if our template's gitignore already has this then this is no problem).

So maybe we should have a `coda add-vs-code-snippets` command or something?

ekoleda-codaio commented:
Fair point, `coda init` has been somewhat agnostic to date (although it does choose `npm` over `yarn` or `pnpm`). What about making it depend on a flag, like `coda init --vscode`? Having an entire CLI command dedicated to copying this file feels like a bit overkill to me personally, and I doubt that people will really ever want to refresh the file.

jonathan-codaio commented:
Well, I'm also thinking about people who already have a pack they've built but now just want to add the snippets. Will running `coda init` in an existing project work ok or will it attempt to overwrite files, or just bail? As long as the use case of "awesome I want these snippets in my existing pack env" is supported, I don't really care what the syntax is.

ekoleda-codaio commented:
I've reverted the changed to `init` and added a new `extensions` command instead:

```
coda extensions <tools..>

Installs developer extensions for working with Packs.

Positionals:
  tools  Which tools to install extensions for.
                            [array] [required] [choices: "vscode"] [default: []]

Options:
  --version  Show version number                                       [boolean]
  --help     Show help                                                 [boolean]

Not enough non-option arguments: got 0, need at least 1
```

jonathan-codaio reviewed (approved):
Sweet
